docs/guides/connecting_to_your_data/overview.md

```diff
@@ -0,0 +1,205 @@
+
+Datasource configurations can be written as YAML files or Python dictionaries.  Regardless of variations due to the underlying data systems, your Datasource's configuration will look roughly like this:
+
+<Tabs
+  groupId="yaml-or-python"
+  defaultValue='yaml'
+  values={[
+  {label: 'YAML', value:'yaml'},
+  {label: 'Python', value:'python'},
+  ]}>
+  <TabItem value="yaml">
+
+```python
+datasource_yaml = fr"""
+name: <name_of_your_datasource>
+class_name: Datasource
+execution_engine:
+    class_name: <class_of_execution_engine>
+data_connectors:
+    <name_of_your_data_connector>:
+        class_name: <class_of_data_connector>
+        <additional_keys_based_on_source_data_system>: <corresponding_values>
+"""
+```
+
+</TabItem>
+<TabItem value="python">
+
+```python
+datasource_config = {
+    "name": "<name_of_your_datasource>",
+    "class_name": "Datasource",
+    "execution_engine": {"class_name": "<class_of_execution_engine>"},
+    "data_connectors": {
+        "<name_of_your_data_connector>": {
+            "class_name": "<class_of_data_connector>",
+            "<additional_keys_based_on_source_data_system>": "<corresponding_values>"
+        }
+    }
+}
+
+```
+
+</TabItem>
+</Tabs>
+
+Please note that this is just a broad outline of the configuration you will be making.  You will find much more detailed examples in our documentation on how to connect to specific source data systems.
+
+The `name` and `class_name` top level keys will be the first you need to define.  The `name` key can be anything you want, but it is best to use a descriptive name as you will use this to reference your Datasource in the future. Unless you are extending Great Expectations and using a subclass of Datasource, you will almost never need to use a `class_name` other than `Datasource` for the top level `class_name` value.
+
+#### Configuring your Datasource's Execution Engine
+
+After your Datasource's configuration has a `name` and `class_name` defined, you will need to define the `execution_engine`.  In your configuration the value of your `execution_engine` will at the very least contain the `class_name` of your Execution Engine, and may also include a `connection_string` if your source data system requires one.
+
+Great Expectations supports Pandas, Spark, and SqlAlchemy as execution engines.
+
+#### Configuring your Datasource's Data Connectors
+
+Great Expectations provides three types of `DataConnector` classes, which are useful in various situations.  Which Data Connector you will want to use will depend on the format of your source data systems.
+- An InferredAssetDataConnector infers the `data_asset_name` by using a regex that takes advantage of patterns that exist in the filename or folder structure.  If your source data system is designed so that it can easily be parsed by regex, this will allow new data to be included by the Datasource automatically.
```

Comment:
I see that this is the current definition in the How to Choose Which Data Connector to Use. That said, while this definition definitely works for data on a file system, you may want to confirm that it is accurate for databases as well.

Reply:
I've updated this section to include the `InferredAssetSqlDataConnector` for SQL based source data systems, and to specify that the `InferredAssetDataConnector` is used for filesystem based source data systems.